TWTValidation/JSON Validator/Parser/TWTJSONSchemaParser.m

```diff
@@ -0,0 +1,652 @@
+//
+//  TWTJSONSchemaParser.m
+//  TWTValidation
+//
+//  Created by Jill Cohen on 12/16/14.
+//  Copyright (c) 2014 Two Toasters, LLC.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+#import <TWTValidation/TWTJSONSchemaParser.h>
+
+#import <TWTValidation/TWTJSONSchemaArrayASTNode.h>
+#import <TWTValidation/TWTJSONSchemaASTNode.h>
+#import <TWTValidation/TWTJSONSchemaBooleanValueASTNode.h>
+#import <TWTValidation/TWTJSONSchemaDepedencyASTNode.h>
+#import <TWTValidation/TWTJSONSchemaGenericASTNode.h>
+#import <TWTValidation/TWTJSONSchemaKeywordConstants.h>
+#import <TWTValidation/TWTJSONSchemaNamedPropertyASTNode.h>
+#import <TWTValidation/TWTJSONSchemaNumberASTNode.h>
+#import <TWTValidation/TWTJSONSchemaObjectASTNode.h>
+#import <TWTValidation/TWTJSONSchemaPatternPropertyASTNode.h>
+#import <TWTValidation/TWTJSONSchemaStringASTNode.h>
+#import <TWTValidation/TWTJSONSchemaTopLevelASTNode.h>
+#import <TWTValidation/TWTJSONSchemaValidTypesConstants.h>
+
+
+static NSString *const kTWTJSONExceptionErrorKey = @"error";
+NSString *const kTWTJSONSchemaParserErrorInfoDomainKey = @"TWTJSONSchemaParserErrorDomain";
+NSString *const kTWTJSONSchemaParserErrorInfoPathKey = @"path";
+NSString *const kTWTJSONSchemaParserErrorInfoInvalidObjectKey = @"object";
```

Comment:
For the string literals, I usually use the whole name of the constant (minus the “key” bit). The thing you have to be concerned about is whether it might collide with another key that is declared elsewhere. Using the whole name basically adds your class prefix to it. That is, it’s pretty easy to imagine some other person using “object” in a dictionary, and one of you might overwrite the other’s value.